test/Test/BaseDAO.hs

```diff
@@ -0,0 +1,271 @@
+-- SPDX-FileCopyrightText: 2020 TQ Tezos
+-- SPDX-License-Identifier: LicenseRef-MIT-TQ
+
+module Test.BaseDAO
+  ( test_BaseDAO
+  ) where
+
+import Universum
+
+import qualified Data.Map.Strict as Map
+import Lorentz
+import Lorentz.Test
+import Morley.Nettest
+import Test.Tasty (TestTree, testGroup)
+import Test.Tasty.HUnit (testCase)
+import Util.Named
+
+import qualified Lorentz.Contracts.Spec.FA2Interface as FA2
+import qualified Indigo.Contracts.SampleFA2 as SampleFA2 -- TODO: Replace SampleFA2 with BaseDAO
+
+-- TODO: Replace SampleFA2 with BaseDAO
+baseDAOStorage
+  :: Address -> Map (Address, FA2.TokenId) Natural
+  -> SampleFA2.Operators -> SampleFA2.Storage
+baseDAOStorage admin bal operators =
+  SampleFA2.mkStorage admin daoToken bal operators
+  where
+    daoToken :: [ FA2.TokenMetadata ]
+    daoToken =
+      [ FA2.TokenMetadata
+          { token_id = 0
+          , symbol = [mt|unfrozen_token|]
+          , name = [mt|Unfrozen Token|]
+          , decimals = 8
+          , extras = Map.empty
+          }
+      , FA2.TokenMetadata
+          { token_id = 1
+          , symbol = [mt|frozen_token|]
+          , name = [mt|Frozen Token|]
+          , decimals = 8
+          , extras = Map.empty
+          }
+      ]
+
+-- TODO: Replace SampleFA2 with BaseDAO
+baseDAOContract :: FA2.PermissionsDescriptor -> Contract FA2.Parameter SampleFA2.Storage
```

Comment:
Why do you still pass `PermissionsDescriptor` here if it's known in advance? :thinking:

Reply:
I forgot this, updated and remove the passed `PermissionsDescriptor`.